### ISSUE TITLE
[Bug][TF FE]: model optimizer error - SSD model

### ISSUE THREAD

**Issue by Hankyu-Kim** [openvinotoolkit/openvino#16389]
# System information (version)
- OpenVINO => 2021.4
- Operating System / Platform => Ubuntu 18.04
- Problem classification: Model Optimizer (.pb to .xml/.bin conversion)
- Framework: TensorFlow (if applicable)
- Model name: ssd mobilenet v1 (if applicable)

### after put the content below

image_resizer { 

      fixed_shape_resizer { 

        height: 360 

        width: 640 

      } 

    } 

I tried 
$ mo_tf.py --input_model /home/park/ssd_mobilenet_v1_all_6c_off_2312_3_5m.pb --input_shape [1,360,640,3] 
on local

### Error message is on down below.

Model Optimizer arguments:
Common parameters:
- Path to the Input Model: 	/home/park/ssd_mobilenet_v1_all_6c_off_2312_3_5m.pb
- Path for generated IR: 	/opt/intel/openvino_2021.4.752/deployment_tools/model_optimizer/.
- IR output name: 	ssd_mobilenet_v1_all_6c_off_2312_3_5m
- Log level: 	ERROR
- Batch: 	Not specified, inherited from the model
- Input layers: 	Not specified, inherited from the model
- Output layers: 	Not specified, inherited from the model
- Input shapes: 	[1,640,640,3]
- Mean values: 	Not specified
- Scale values: 	Not specified
- Scale factor: 	Not specified
- Precision of IR: 	FP32
- Enable fusing: 	True
- Enable grouped convolutions fusing: 	True
- Move mean values to preprocess section: 	None
- Reverse input channels: 	False
TensorFlow specific parameters:
- Input model in text protobuf format: 	False
- Path to model dump for TensorBoard: 	None
- List of shared libraries with TensorFlow custom layers implementation: 	None
- Update the configuration file with input/output node names: 	None
- Use configuration file used to generate the model with Object Detection API: 	None
- Use the config file: 	None
- Inference Engine found in: 	/opt/intel/openvino_2021/python/python3.7/openvino
Inference Engine version: 	2021.4.2-3974-e2a469a3450-releases/2021/4
Model Optimizer version: 	2021.4.2-3974-e2a469a3450-releases/2021/4
2023-03-14 14:20:09.229417: I tensorflow/core/platform/cpu_feature_guard.cc:193] This TensorFlow binary is optimized with oneAPI Deep Neural Network Library (oneDNN) to use the following CPU instructions in performance-critical operations:  AVX2 FMA
To enable them in other operations, rebuild TensorFlow with the appropriate compiler flags.
2023-03-14 14:20:10.019048: W tensorflow/compiler/xla/stream_executor/platform/default/dso_loader.cc:64] Could not load dynamic library 'libcudart.so.11.0'; dlerror: libcudart.so.11.0: cannot open shared object file: No such file or directory; LD_LIBRARY_PATH: /opt/intel/openvino_2021/data_processing/dl_streamer/lib:/opt/intel/openvino_2021/data_processing/gstreamer/lib:/opt/intel/openvino_2021/opencv/lib:/opt/intel/openvino_2021/deployment_tools/ngraph/lib:/opt/intel/openvino_2021/deployment_tools/inference_engine/external/tbb/lib::/opt/intel/openvino_2021/deployment_tools/inference_engine/external/hddl/lib:/opt/intel/openvino_2021/deployment_tools/inference_engine/external/omp/lib:/opt/intel/openvino_2021/deployment_tools/inference_engine/external/gna/lib:/opt/intel/openvino_2021/deployment_tools/inference_engine/external/mkltiny_lnx/lib:/opt/intel/openvino_2021/deployment_tools/inference_engine/lib/intel64
2023-03-14 14:20:10.019161: I tensorflow/compiler/xla/stream_executor/cuda/cudart_stub.cc:29] Ignore above cudart dlerror if you do not have a GPU set up on your machine.
2023-03-14 14:20:11.806258: W tensorflow/compiler/xla/stream_executor/platform/default/dso_loader.cc:64] Could not load dynamic library 'libnvinfer.so.7'; dlerror: libnvinfer.so.7: cannot open shared object file: No such file or directory; LD_LIBRARY_PATH: /opt/intel/openvino_2021/data_processing/dl_streamer/lib:/opt/intel/openvino_2021/data_processing/gstreamer/lib:/opt/intel/openvino_2021/opencv/lib:/opt/intel/openvino_2021/deployment_tools/ngraph/lib:/opt/intel/openvino_2021/deployment_tools/inference_engine/external/tbb/lib::/opt/intel/openvino_2021/deployment_tools/inference_engine/external/hddl/lib:/opt/intel/openvino_2021/deployment_tools/inference_engine/external/omp/lib:/opt/intel/openvino_2021/deployment_tools/inference_engine/external/gna/lib:/opt/intel/openvino_2021/deployment_tools/inference_engine/external/mkltiny_lnx/lib:/opt/intel/openvino_2021/deployment_tools/inference_engine/lib/intel64
2023-03-14 14:20:11.806424: W tensorflow/compiler/xla/stream_executor/platform/default/dso_loader.cc:64] Could not load dynamic library 'libnvinfer_plugin.so.7'; dlerror: libnvinfer_plugin.so.7: cannot open shared object file: No such file or directory; LD_LIBRARY_PATH: /opt/intel/openvino_2021/data_processing/dl_streamer/lib:/opt/intel/openvino_2021/data_processing/gstreamer/lib:/opt/intel/openvino_2021/opencv/lib:/opt/intel/openvino_2021/deployment_tools/ngraph/lib:/opt/intel/openvino_2021/deployment_tools/inference_engine/external/tbb/lib::/opt/intel/openvino_2021/deployment_tools/inference_engine/external/hddl/lib:/opt/intel/openvino_2021/deployment_tools/inference_engine/external/omp/lib:/opt/intel/openvino_2021/deployment_tools/inference_engine/external/gna/lib:/opt/intel/openvino_2021/deployment_tools/inference_engine/external/mkltiny_lnx/lib:/opt/intel/openvino_2021/deployment_tools/inference_engine/lib/intel64
2023-03-14 14:20:11.806443: W tensorflow/compiler/tf2tensorrt/utils/py_utils.cc:38] TF-TRT Warning: Cannot dlopen some TensorRT libraries. If you would like to use Nvidia GPU with TensorRT, please make sure the missing libraries mentioned above are installed properly.
[ WARNING ]
Detected not satisfied dependencies:
tensorflow: installed: 2.11.0, required: ~= 2.4.1
numpy: installed: 1.21.6, required: < 1.20
Please install required versions of components or use install_prerequisites script
/opt/intel/openvino_2021.4.752/deployment_tools/model_optimizer/install_prerequisites/install_prerequisites_tf2.sh
Note that install_prerequisites scripts may install additional components.
/opt/intel/openvino_2021/deployment_tools/model_optimizer/mo/front/tf/common.py:10: DeprecationWarning: np.bool is a deprecated alias for the builtin bool. To silence this warning, use bool by itself. Doing this will not modify any behavior and is safe. If you specifically wanted the numpy scalar type, use np.bool_ here.
Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations
_tf_np_mapping = [('DT_BOOL', np.bool, lambda pb: pb.bool_val, lambda x: bool_cast(x)),
/opt/intel/openvino_2021/deployment_tools/model_optimizer/mo/front/tf/common.py:22: DeprecationWarning: np.str is a deprecated alias for the builtin str. To silence this warning, use str by itself. Doing this will not modify any behavior and is safe. If you specifically wanted the numpy scalar type, use np.str_ here.
Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations
('DT_STRING', np.str, lambda pb: pb.string_val, lambda x: np.str(x)),
/opt/intel/openvino_2021/deployment_tools/model_optimizer/mo/front/tf/extractors/utils.py:64: DeprecationWarning: np.str is a deprecated alias for the builtin str. To silence this warning, use str by itself. Doing this will not modify any behavior and is safe. If you specifically wanted the numpy scalar type, use np.str_ here.
Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations
if type_helper[0] != np.str:
/opt/intel/openvino_2021/deployment_tools/model_optimizer/extensions/analysis/boolean_input.py:13: DeprecationWarning: np.bool is a deprecated alias for the builtin bool. To silence this warning, use bool by itself. Doing this will not modify any behavior and is safe. If you specifically wanted the numpy scalar type, use np.bool_ here.
Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations
nodes = graph.get_op_nodes(op='Parameter', data_type=np.bool)
/opt/intel/openvino_2021/deployment_tools/model_optimizer/extensions/ops/select.py:50: DeprecationWarning: np.bool is a deprecated alias for the builtin bool. To silence this warning, use bool by itself. Doing this will not modify any behavior and is safe. If you specifically wanted the numpy scalar type, use np.bool_ here.
Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations
output_value = np.array(output_value, dtype=resulting_tensors[not np.bool(condition_value.item(0))].dtype)
/opt/intel/openvino_2021/deployment_tools/model_optimizer/mo/front/common/partial_infer/concat.py:36: DeprecationWarning: np.bool is a deprecated alias for the builtin bool. To silence this warning, use bool by itself. Doing this will not modify any behavior and is safe. If you specifically wanted the numpy scalar type, use np.bool_ here.
Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations
mask = np.zeros_like(shape, dtype=np.bool)

[ ERROR ]  Cannot infer shapes or values for node Postprocessor/BatchMultiClassNonMaxSuppression/map/while/Slice_1".
[ ERROR ]  The non-constant start/end values for Slice operation "Postprocessor/BatchMultiClassNonMaxSuppression/map/while/Slice_1" are not supported
[ ERROR ]
[ ERROR ]  It can happen due to bug in custom shape infer function <function Slice.infer at 0x7fa0e1295c20>.
[ ERROR ]  Or because the node inputs have incorrect values/shapes.
[ ERROR ]  Or because input shapes are incorrect (embedded to the model or passed via --input_shape).
[ ERROR ]  Run Model Optimizer with --log_level=DEBUG for more information.

[ ANALYSIS INFO ]  Your model looks like TensorFlow Object Detection API Model.
Check if all parameters are specified:
--tensorflow_use_custom_operations_config
--tensorflow_object_detection_api_pipeline_config
--input_shape (optional)
--reverse_input_channels (if you convert a model to use with the Inference Engine sample applications)

Detailed information about conversion of this model can be found at
https://docs.openvinotoolkit.org/latest/_docs_MO_DG_prepare_model_convert_model_tf_specific_Convert_Object_Detection_API_Models.html

[ ERROR ]  Exception occurred during running replacer "REPLACEMENT_ID" (<class 'extensions.middle.PartialInfer.PartialInfer'>): Stopped shape/value propagation at "Postprocessor/BatchMultiClassNonMaxSuppression/map/while/Slice_1" node.
For more information please refer to Model Optimizer FAQ, question #38. (https://docs.openvinotoolkit.org/latest/openvino_docs_MO_DG_prepare_model_Model_Optimizer_FAQ.html?question=38#question-38)[](url)

**Comment by Hankyu-Kim** [comment]
[ssd_mobilenet_v1_all_6c_off_2312_3_5m.zip](https://github.com/openvinotoolkit/openvino/files/11015639/ssd_mobilenet_v1_all_6c_off_2312_3_5m.zip)

**Comment by Hankyu-Kim** [comment]
https://drive.google.com/file/d/1vTo1E6sACpYuWWMe82MiZ_cV2VjtTunH/view?usp=sharing

**Comment by rkazants** [comment]
Hi @Hankyu-Kim,

You have TF1 OD model for which we require special conversion procedure using config files described here: https://docs.openvino.ai/latest/openvino_docs_MO_DG_prepare_model_convert_model_tf_specific_Convert_Object_Detection_API_Models.html.

For this you need to have `pipeline.config` file. Can you provide it please?

Best regards,
Roman

**Comment by Hankyu-Kim** [comment]
[ssd_mobilenet_v1_techpol_1B05.zip](https://github.com/openvinotoolkit/openvino/files/11016012/ssd_mobilenet_v1_techpol_1B05.zip)

**Comment by Hankyu-Kim** [comment]
https://drive.google.com/file/d/14OSbd7yka4LfFJUych9s4NElq6eO3VxK/view?usp=share_link

here it is! Thank you @rkazants !! you re my savior

**Comment by Hankyu-Kim** [comment]
also, I wish I can know which tensorflow version (1 or 2 ) that random .pb file used too~ How did you know mine was TF1 OD model?

**Comment by Hankyu-Kim** [comment]
Oh! Hi Roman Kazantsev,

I uploaded it on our conversation on website : https://github.com/openvinotoolkit/openvino/issues/16389
Just in case I will attach my .config file on here too!
Thank you always!

Best regards,
Hankyu Kim

-----Original Message-----

**Comment by rkazants** [comment]
Hi @Hankyu-Kim, I did try with `pipeline.config` file you provided but the MO tool says that the file is incorrect. Could you please check if you have changed something in the file?

The idea is to convert this model using the transformation config file like described here: https://docs.openvino.ai/latest/openvino_docs_MO_DG_prepare_model_convert_model_tf_specific_Convert_Object_Detection_API_Models.html.

Best regards,
Roman

**Comment by andrei-kochin** [comment]
@Hankyu-Kim could you please update on Roman's question?

**Comment by Hankyu-Kim** [comment]
I guess it started from file name 'ssd_mobilenet_v1_all_1c_0528.pb', I though config file doesn't matter when you train with same model, here is zip file of the first model, I'll also try with all of these with command from the refernece site that you gave me. Like this : mo --input_model=/tmp/ssd_inception_v2_coco_2018_01_28/frozen_inference_graph.pb --transformations_config front/tf/ssd_v2_support.json --tensorflow_object_detection_api_pipeline_config /tmp/ssd_inception_v2_coco_2018_01_28/pipeline.config --reverse_input_channels

![Screenshot from 2023-04-14 09-24-11](https://user-images.githubusercontent.com/108254705/231910719-1b198248-6bf3-4026-bc18-b845ca321bd1.png)
[ssd_mobilenet_v1_all_1c_0528.zip](https://github.com/openvinotoolkit/openvino/files/11227578/ssd_mobilenet_v1_all_1c_0528.zip)

**Comment by akladiev** [comment]
This issue will be closed in 2 weeks in case of no activity.

**Comment by avitial** [comment]
Ref. 116068

**Comment by rkazants** [comment]
@Hankyu-Kim, please check the fix. Now your models should be converted out-of-the-box just by running: mo --input_model model_name.pb.
Please make a note that this fix will be available in the next release 2023.3 LST or next nightly build.
